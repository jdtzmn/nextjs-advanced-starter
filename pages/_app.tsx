import App, { Container } from 'next/app'
import React from 'react'
import withReduxStore from '../lib/with-redux-store'
import { Provider } from 'react-redux'
import { Store } from 'redux'

interface MyAppProps {
  store: Store
}

class MyApp extends App<MyAppProps> {
  render () {
    const { Component, pageProps, store } = this.props
    return (
      <Container>
        <Provider store={store}>
          <Component {...pageProps} />
        </Provider>
      </Container>
    )
  }
}

export default withReduxStore(MyApp)
