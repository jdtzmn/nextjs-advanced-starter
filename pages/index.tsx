import * as React from 'react'
import { Store } from 'redux'
import styled from 'styled-components'
import { connectIndexContainer } from '../containers'

const Title = styled.h1`
  padding-top: 50px;
  text-align: center;
`

const Count = styled.p`
  font-size: 20px;
  text-align: center;
`
Count.displayName = 'count'

const Button = styled.button`
  color: white;
  background: rgb(0, 122, 255);
  display: block;
  margin: 0 auto;
  padding: 16px;
  font-size: 16px;
  outline: none;
  border: 0;
  border-radius: 4px;

  &:hover {
    background: rgba(0, 122, 255, 0.8);
  }
`
Button.displayName = 'incrementButton'

export interface IndexProps {
  count: number,
  increment: () => any,
  store: Store,
  theme: object
}

export class Index extends React.Component<IndexProps> {
  render () {
    const { count, increment } = this.props

    return (
      <div>
        <Title>nextjs-advanced-starter</Title>
        <Count>{count}</Count>
        <Button onClick={increment}>Click To Increment</Button>
      </div>
    )
  }
}

export default connectIndexContainer(Index)
