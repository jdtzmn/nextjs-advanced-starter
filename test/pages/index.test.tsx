import * as React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'

import { Index } from 'pages/index'
import createStore from '@/store'

const store = createStore()

describe('Index', () => {
  const mockIncrement = jest.fn()
  let wrapper: ShallowWrapper
  beforeEach(() => {
    wrapper = shallow(<Index
      count={0}
      increment={mockIncrement}
      store={store}
      theme={{}}
    />)
  })

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should call increment on button click', () => {
    wrapper
      .find('incrementButton')
      .simulate('click')
    expect(mockIncrement).toHaveBeenCalled()
  })

  it('should render the count number', () => {
    wrapper.setProps({ count: 1 })
    expect(wrapper.find('count').text()).toBe('1')
  })
})
