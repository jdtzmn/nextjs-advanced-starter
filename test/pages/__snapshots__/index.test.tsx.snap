// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index should render correctly 1`] = `
<div>
  <styled.h1>
    nextjs-advanced-starter
  </styled.h1>
  <count>
    0
  </count>
  <incrementButton
    onClick={[MockFunction]}
  >
    Click To Increment
  </incrementButton>
</div>
`;
